viam-orb-slam3.go

```diff
@@ -488,33 +466,31 @@ func NewBuiltIn(ctx context.Context, deps registry.Dependencies, config config.S
 	}
 
 	modelName := string(config.Model.Name)
-	slamLib, ok := slam.SLAMLibraries[modelName]
 	if !ok {
 		return nil, errors.Errorf("%v algorithm specified not in implemented list", modelName)
 	}
 
-	slamMode, ok := slamLib.SlamMode[svcConfig.ConfigParams["mode"]]
-	if !ok {
+	subAlgo := OrbslamAlgorithm(svcConfig.ConfigParams["mode"])
+	validAlgos := []OrbslamAlgorithm{Mono, Rgbd}
+	if !slices.Contains(validAlgos, subAlgo) {
 		return nil, errors.Errorf("getting data with specified algorithm %v, and desired mode %v",
 			modelName, svcConfig.ConfigParams["mode"])
 	}
 
 	slamConfig.SetupDirectories(svcConfig.DataDirectory, logger)
 
-	if slamMode == slam.Rgbd || slamMode == slam.Mono {
-		var directoryNames []string
-		if slamMode == slam.Rgbd {
-			directoryNames = []string{"rgb", "depth"}
-		} else if slamMode == slam.Mono {
-			directoryNames = []string{"rgb"}
-		}
-		for _, directoryName := range directoryNames {
-			directoryPath := filepath.Join(svcConfig.DataDirectory, "data", directoryName)
-			if _, err := os.Stat(directoryPath); os.IsNotExist(err) {
-				logger.Warnf("%v directory does not exist", directoryPath)
-				if err := os.Mkdir(directoryPath, os.ModePerm); err != nil {
-					return nil, errors.Errorf("issue creating directory at %v: %v", directoryPath, err)
-				}
```

Comment:
We have already checked to ensure it is a valid orbslam algorithm and we can remove one layer of if statements.